src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.html

```diff
@@ -252,6 +252,132 @@
+                      <mat-label>{{'settings.pricing.restriction_min_energy_unit' | translate}}</mat-label>
+                      <input matInput type="text" placeholder="{{'settings.pricing.restriction_min_energy_unit' | translate}}"
+                      [formControl]="minEnergyKWhValue" required/>
+                      <mat-error *ngIf="minEnergyKWhValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                      <mat-error *ngIf="minEnergyKWhValue?.errors?.pattern">{{'general.error_number_pattern' | translate}}</mat-error>
+                    </mat-form-field>
+                  </div>
+                  <!-- toggle max energy -->
+                  <div class="col-md-2 offset-1 pricing-dimension">
+                    <mat-slide-toggle (change)="toggle($event)" id="maxEnergyKWh" [checked]="maxEnergyKWhEnabled.value">
+                      {{'settings.pricing.restriction_max_energy' | translate}}
+                    </mat-slide-toggle>
+                  </div>
+                  <!-- value max energy -->
+                  <div class="col-md-2">
+                    <mat-form-field *ngIf="maxEnergyKWhEnabled.value">
+                      <mat-label>{{'settings.pricing.restriction_max_energy_unit' | translate}}</mat-label>
+                      <input matInput type="text" placeholder="{{'settings.pricing.restriction_max_energy_unit' | translate}}"
+                      [formControl]="maxEnergyKWhValue" required/>
+                      <mat-error *ngIf="maxEnergyKWhValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                      <mat-error *ngIf="maxEnergyKWhValue?.errors?.pattern">{{'general.error_number_pattern' | translate}}</mat-error>
+                    </mat-form-field>
+                  </div>
+                </div>
+                <!-- time range -->
+                <div class="row">
+                  <div class="col-md-2 pricing-dimension">
+                    <mat-slide-toggle (change)="toggle($event)" id="timeFrom" [checked]="timeFromEnabled.value">
+                      {{'settings.pricing.restriction_time_range' | translate}}
+                    </mat-slide-toggle>
+                  </div>
+                  <!-- start time -->
+                  <div class="col-md-2">
+                    <mat-form-field *ngIf="timeFromEnabled.value">
+                      <mat-label>{{'settings.pricing.restriction_start_time' | translate}}</mat-label>
+                      <input matInput type="time" id="timeFromValue" name="timeFromValue" [formControl]="timeFromValue" (change)="setMinTime($event)" required/>
+                    </mat-form-field>
+                    <mat-error *ngIf="timeFromValue?.errors?.required">{{'general.mandatory_field' | translate}}</mat-error>
+                    <mat-error *ngIf='timeToValue?.errors?.incorrectEndTime'>{{'settings.pricing.pricing_definition_time_error' | translate}}</mat-error>
```

Comment:
Why not incorrectStartTime here ?

Reply:
And I don't know when this is displayed as you change it to the right value after entering an out of range time.

Reply:
I display the 'timeRangeError' error message now